lib/adapters/test/testplane.ts

```diff
@@ -0,0 +1,54 @@
+import {TestplaneTestResultAdapter} from '../test-result/testplane';
+import {TestStatus, UNKNOWN_ATTEMPT} from '../../constants';
+
+import type {TestAdapter} from './';
+import type {Test, Suite} from 'testplane';
+import type {ReporterTestResult} from '../test-result';
+
+export class TestplaneTestAdapter implements TestAdapter {
+    private _test: Test;
+
+    static create<T extends TestplaneTestAdapter>(this: new (test: Test) => T, test: Test): T {
+        return new this(test);
+    }
+
+    constructor(test: Test) {
+        this._test = test;
+    }
+
+    get originalTest(): Test {
+        return this._test;
+    }
+
+    get id(): string {
+        return this._test.id;
+    }
+
+    get pending(): boolean {
+        return this._test.pending;
+    }
+
+    get disabled(): boolean {
+        return this._test.disabled;
+    }
+
+    get browserId(): string {
+        return this._test.browserId;
+    }
+
+    fullTitle(): string {
+        return this._test.fullTitle();
+    }
+
+    clone(): TestplaneTestAdapter {
```

Comment:
Should it really be inside TestAdapter interface and be implemented by all consumers instead of being standalone pure function?

I like to think of these objects as slim, simple and immutable data objects on which we can apply various transformations via pure functions.

For instance, why would you need to implement clone in each adapter if all information you need is exposed via public props and you could just have a single external clone function which could clone any adapter.

Not critical

Reply:
> Should it really be inside TestAdapter interface and be implemented by all consumers instead of being standalone pure function?

Runnables inside testplane has `clone` method - https://github.com/gemini-testing/testplane/blob/master/src/test-reader/test-object/test.ts#L23-L30 which must be used to correctly clone instance. For other test adapters I can use default `cloneDeep` from lodash in base class in order to not implement it for each tool.